deepinv/training_utils.py

```diff
@@ -316,37 +222,159 @@ def train(
                 for i in range(len(vis_array)):
                     vis_array[i] = rescale_img(vis_array[i], rescale_mode="min_max")
                 grid_image = torchvision.utils.make_grid(vis_array, nrow=y.shape[0])
-                if epoch % freq_plot == 0:
+                if (epoch+1) % self.freq_plot == 0:
                     images = wandb.Image(
                         grid_image,
                         caption=caption,
                     )
                     log_dict_post_epoch["Training samples"] = images
 
-        if wandb_vis:
+        if self.wandb_vis:
             wandb.log(log_dict_post_epoch)
 
-        loss_history.append(total_loss.avg)
-
-        if scheduler:
-            scheduler.step()
-
-        # Saving the model
-        save_model(
-            epoch,
-            model,
-            optimizer,
-            ckp_interval,
-            epochs,
-            loss_history,
-            str(save_path),
-            eval_psnr=eval_psnr if perform_eval else None,
-        )
+    def batch_eval(self, x, x_net):
```

Comment:
not sure if `batch_eval` as such is the best name (for me it would refer to doing forward in eval mode). What about `batch_eval_metric` instead?